### ISSUE TITLE
Engine assumes Rails is using Sprockets

### ISSUE THREAD

**Issue by kodemunki** [revelrylabs/execjs-rails#15]
I don't use Sprockets at all in any of my projects. Looking at the `engine.rb` source I'm assuming `execjs_rails` can only be used when Rails is configured to use Sprockets and serve assets?

**Comment by jwietelmann** [comment]
That is correct. I would certainly be open to pull requests that broke out the sprockets-specific bits into an adapter and then added a separate adapter for the way you want to use it. We just personally use Sprockets with this, so we may not be the best in terms of driving the decision for how it should operate with alternatives. I'm not opposed to writing some of the code myself; I just don't have the right context to make a good decision for you.

tl;dr - yes it's sprockets-only, but I'm willing to change that if you're willing to PR me a sketch of how you'd like it to work

**Comment by jwietelmann** [comment]
I should add that you could specify your own `config.execjs_rails.build_source` lambda in an initializer and that should take care of it. I think that's the only sprockets-specific bit.
